package/remote_package.md

```diff
@@ -1410,3 +1416,229 @@ For CMake 3.19 and later which has JSON support, `xrepo_package` parses the JSON
 output. For previous version of CMake, `xrepo_package` uses only the `--cflags` option
 to get package include directory. Library and cmake module directory are infered from that
 directory, so it maybe unreliable to detect the correct paths.
+
+---
+
+#### Step by Step Remote Packaging Tutorial
+
+---
+
+##### Introduction
+
+###### What is a remote package?
+
+A remote package is a package that is compiled from source
+
+###### Why should I use remote packages?
+
+If you're developing cross-platform libraries, remote packages allow you to avoid manually compiling for every library and platform
+
+###### How can I create remote packages?
+
+1. Upload your source code to a repository
+
+2. In a **separate repository**, create a remote package manifest that points to your source repository
+
+###### How can I consume remote packages?
+
+You need to add the repository that contains the remote package manifest to your project, find the package then and add it to your desired target.
+
+##### Example
+
+In this example, we’ll create a remote package for a static library named `foo`, built using **Windows with MSVC**, and then **consume it on MSYS2 using Clang**.
+
+- Create an xmake project
+  
+  ```powershell
+  xmake create -P package_remote_origin
+  ```
+
+- Imitate this filetree to prepare files for your package
+  
+  ```powershell
+  │   .gitignore
+  │   xmake.lua
+  │
+  └───src
+      │   main.cpp
+      │
+      ├───inc
+      │   └───foo
+      │           foo.hpp
+      │
+      └───lib
+          └───foo
+                  foo.cpp
+  ```
+
+- Create static library target in xmake
+  
+  ```lua
+  target("foo")
+      set_kind("static")
+      add_files("src/lib/foo/*.cpp")
+      add_headerfiles("src/inc/foo/*.hpp")
+      add_includedirs("src/inc/foo", {public = true})
+  ```
+
+- Implement the functionality of your target
+  
+  foo.hpp
+  
+  ```cpp
+  void foo();
+  ```
+  
+  foo.cpp
+  
+  ```cpp
+  #include <iostream>
+  #include "foo.hpp"
+  
+  void foo()
+  {
+      std::cout << "foo";
+  }
+  ```
+
+- Build your project to see if it compiles
+  
+  ```powershell
+  xmake build
+  ```
+
+- Create a source repository for your package with a version tag
+  
+  ```powershell
+  git init
+  ```
+  
+  ```powershell
+  git checkout -b package_source
+  ```
+  
+  ```powershell
+  git add .\src\
+  ```
+  
+  ```powershell
+  git add xmake.lua
+  ```
+  
+  ```powershell
+  git commit -m "init"
+  ```
+  
+  ```powershell
+  git tag v1.0.0 
+  ```
+  
+  ```powershell
+  gh repo create xmake_remote_package_tutorial_source --public
+  ```
+  
+  ```powershell
+  git remote add source https://github.com/mccakit/xmake_remote_package_tutorial_source.git
+  ```
+  
+  ```powershell
+  git push -u source package_source v1.0.0
```

Comment:
you can check code format. https://github.com/xmake-io/xmake-docs/commit/aadf1a64542f45a91acf94370aeac1715e2d2a09#diff-aba85b8c1a63b4f97272cfc680c7d64d9203e5db9fd8e4cd12905637428e7543